File: Code/CH5 Code/C#/MovieList-c/ListViewSorter.cs
using System;
using System.Windows.Forms;
using System.Collections;

namespace MovieList_c
{
	/// <summary>
	/// This class sorts a ListView control by column
	/// </summary>
	public class ListViewSorter: IComparer
	{
    private int       mCol;
    private SortOrder mOrder;

    public ListViewSorter(int column, SortOrder order) 
    {
      mCol=column;
      mOrder = order;
    }
    public int Compare(object x, object y) 
    {
      int returnVal = String.Compare(((ListViewItem)x).SubItems[mCol].Text,
        ((ListViewItem)y).SubItems[mCol].Text);
      if(mOrder == SortOrder.Descending)
        return (returnVal *= -1);
      else
        return returnVal;
    }
  }

}

File: Code/CH5 Code/VB/MovieList-vb/ListViewSorter.vb
Option Strict On
Imports System.Windows.Forms
Imports System.Collections

Public Class ListViewSorter
  Implements IComparer

  Dim mCol As Integer
  Dim mOrder As SortOrder

  Public Sub New(ByVal column As Integer, ByVal order As SortOrder)
    mCol = column
    mOrder = order
  End Sub

  Public Function Compare(ByVal x As Object, ByVal y As Object) As Integer _
                          Implements System.Collections.IComparer.Compare
    Dim returnVal As Integer = String.Compare( _
                                (CType(x, ListViewItem)).SubItems(mCol).Text, _
                                (CType(y, ListViewItem).SubItems(mCol).Text))
    If mOrder = SortOrder.Descending Then
      returnVal *= -1
      Return returnVal
    Else
      Return returnVal
    End If
  End Function

End Class
